### PR TITLE
UIEH-166: Support user selection of proxy on editing a package

### PR DESCRIPTION
**Pull request by keathleydavidj**
## Purpose
To resolve: [UIEH-166](https://issues.folio.org/browse/UIEH-166)
A user should be able to set their proxy at the package level so that all titles in that package inherit the same proxy. We've decided to support this functionality by introducing a simple `<Select />` element labelled `Proxy` to the package edit view.

## Approach
- [x] Fetch relevant data in the route (Proxy types, Provider proxy)
- [x] Display current value of a packages' proxy on the show view
- [x] Add the select box for assigning proxy value to the edit view
- [x] Support inheritance of proxies from provider level and prepend `inherited` appropriately in both package-show and package-edit
- [x] Associated tests

## Screenshots
package_proxy

### CODE REVIEW

File: src/components/package/edit-managed/managed-package-edit.js [hunk 1]
```diff
@@ -311,6 +314,10 @@ class ManagedPackageEdit extends Component {
                           </label>
                         )}
                       </div>
+                      <ProxySelectField
+                        proxy={model.proxy}
+                        proxyTypes={proxyTypes}
+                      />
```
Comment (sduvvuri-ebsco) [inline src/components/package/edit-managed/managed-package-edit.js]: Around this, you probably need to add a conditional like this
```{(!proxyTypes.request.isResolved) ? (
                         <Icon icon="spinner-ellipsis" />
                      ) : (
                          <ProxySelectField proxy={model.proxy} proxyTypes={proxyTypes} />
                      )}

Looks like there is delay in the model resolving when your dropdown is showing empty list

File: src/components/provider/_fields/proxy-select/proxy-select-field.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ function ProxySelectField({ proxyTypes, rootProxy, intl }) {
         name="proxyId"
         component={Select}
         dataOptions={options}
-        label={intl.formatMessage({ id: 'ui-eholdings.provider.proxy' })}
+        label={intl.formatMessage({ id: 'ui-eholdings.proxy' })}
```
Comment (SamKeathley) [inline src/components/provider/_fields/proxy-select/proxy-select-field.js]: Thanks for fixing this for me!

File: src/routes/package-edit.js [hunk 3]
```diff
@@ -117,6 +128,11 @@ class PackageEditRoute extends Component {
         model.contentType = values.contentType;
       }
 
+      if ('proxyId' in values) {
+        model.proxy.id = values.proxyId;
+        model.proxy.inherited = false;
+      }
+
```
Comment (carolegodfrey) [inline src/routes/package-edit.js]: Really minor thing in packageEditSubmitted function -- may want to reset `model.proxy = {}` when package is being de-selected

Comment (sduvvuri-ebsco) [reply src/routes/package-edit.js]: Addressed 👍

**Pull request merged**